src/rdrand.rs

```diff
@@ -12,7 +12,10 @@ use crate::util::LazyBool;
 use crate::Error;
 use core::arch::x86_64::_rdrand64_step;
 use core::mem;
-use core::num::NonZeroU32;
+
+const NO_RDRAND: u16 = 0;
+const BAD_RDRAND: u16 = 1;
+const FAILED_RDRAND: u16 = 2;
```

Comment:
Re-using the same custom codes across multiple platforms doesn't seem useful; it's not like we're short of them. I think it would be better to have a single list of all custom codes in the `error` module?

This implies we could merge all `custom_description` implementations — although that has the disadvantage of defining all error strings in all binaries, so we might also prefer not to do this. (Either way I think the codes should be distinct.)

Reply:
I moved all the error constants to `error.rs` and now have a single `internal_desc` function to get the description for each error constant.

I don't think we should worry about a few more string constants in static memory (it's only like O(100) bytes). If a user really cares, they can just remove any calls to `Debug` or `Display` the `Error`, and then LTO will get rid of the strings.